platform/kobo/koreader.sh

```diff
@@ -611,10 +611,20 @@ if [ ${RETURN_VALUE} -ne ${KO_RC_HALT} ]; then
         # if we were called from advboot then we must reboot to go to the menu
         # NOTE: This is actually achieved by checking if KSM or a KSM-related script is running:
         #       This might lead to false-positives if you use neither KSM nor advboot to launch KOReader *without nickel running*.
-        if ! pgrep -f kbmenu >/dev/null 2>&1; then
+        if ! pkill -0 kbmenu; then
             /sbin/reboot
         fi
     fi
+else
+    if [ "${VIA_NICKEL}" = "false" ]; then
+        if pkill -0 kbmenu; then
+            # If we were started by KSM and requested an exit, do *NOT* exit the script so as not to re-enter KSM at all,
+            # to make sure the device powers off with our own ScreenSaver displayed.
+            while :; do
+                sleep 2073600
+            done
```

Comment:
Er, this seems a bit excessive in every way imaginable?

Reply:
An endless while loop coupled with a sleep of hundreds of hours?! lol

Reply:
I'm thinking a simple 5-10 seconds (make it 30 or 60 if you like) without any while loops would do the trick. ;-)

Reply:
I know it's supposed to power off, but on the off chance it *doesn't* you're making it a hundred times harder to figure out that something went wrong, and worst case scenario it might drain the battery.

Reply:
Hence *not* doing that is significantly more harmless. There's no upside to value higher than 10s, imho.

Reply:
Fair enough.